src/nifc/codegen.nim

```diff
@@ -350,11 +350,26 @@ proc genVarDecl(c: var GeneratedCode; t: Tree; n: NodePos; vk: VarKind; toExtern
     if vis == StaticC:
       c.code.insert(Token(StaticKeyword), beforeDecl)
     if t[d.value].kind != Empty:
-      c.add AsgnOpr
-      if vk != IsLocal: inc c.inSimpleInit
-      genx c, t, d.value
-      if vk != IsLocal: dec c.inSimpleInit
-    c.add Semicolon
+      if vk == IsGlobal and
+        t[d.value].kind notin {IntLit, UIntLit, FloatLit,
+          CharLit, StrLit, FalseC, TrueC, AconstrC,
+          OconstrC, InfC, NegInfC, NanC, SufC}:
```

Comment:
You need some `isLiteral` deep check for OconstrC and AconstrC as `[1, 2]` is very different from `[myVar, 3]`.

Reply:
I have no clues how to deal with `Tree` iteratively. Do I need to use `hasNext` to query the end of the tree or something?

Reply:
You use the simple `while n.kind != ParRi:...` loop but your way is fine too. Also NIFC doesn't yet use the newer APIs so that's probably why you were confused.